packages/@jsii/python-runtime/tests/test_runtime_type_checking.py

```diff
@@ -1,14 +1,31 @@
 import pytest
 import re
-from typing import Optional
+from typing import Optional, cast, Any
 
 from scope.jsii_calc_lib.custom_submodule_name import NestingClass
 from scope.jsii_calc_lib import Number
 import jsii_calc
 import jsii
+import typeguard
 
+from importlib.metadata import version
+
+TYPEGUARD_MAJOR_VERSION = int(version("typeguard").split(".")[0])
+
+
+@jsii.implements(jsii_calc.IBellRinger)
+class PythonInvalidBellRinger:
```

Comment:
Thanks for 4.x we can now statically typecheck protocols so added a test for it.